HDDS-1491. Ozone KeyInputStream seek() should not read the chunk file.

hanishakoneru opened the pull request.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 61 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 470 | trunk passed |
| +1 | compile | 204 | trunk passed |
| +1 | checkstyle | 56 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 880 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 126 | trunk passed |
| 0 | spotbugs | 254 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 452 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 413 | the patch passed |
| +1 | compile | 218 | the patch passed |
| +1 | javac | 218 | the patch passed |
| +1 | checkstyle | 60 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 1 | The patch has no whitespace issues. |
| +1 | shadedclient | 664 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 126 | the patch passed |
| +1 | findbugs | 439 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 164 | hadoop-hdds in the patch failed. |
| -1 | unit | 922 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 36 | The patch does not generate ASF License warnings. |
| | | 5436 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.pipeline.TestRatisPipelineUtils |
|   | hadoop.hdds.scm.pipeline.TestNodeFailure |
|   | hadoop.ozone.om.TestOMDbCheckpointServlet |
|   | hadoop.ozone.TestMiniChaosOzoneCluster |
|   | hadoop.ozone.om.TestOmBlockVersioning |
|   | hadoop.ozone.om.TestOmInit |
|   | hadoop.hdds.scm.container.TestContainerStateManagerIntegration |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.scm.pipeline.TestSCMPipelineMetrics |
|   | hadoop.ozone.client.rpc.TestCommitWatcher |
|   | hadoop.ozone.om.TestOmMetrics |
|   | hadoop.ozone.om.TestMultipleContainerReadWrite |
|   | hadoop.ozone.client.rpc.TestContainerStateMachineFailures |
|   | hadoop.ozone.client.rpc.TestOzoneClientRetriesOnException |
|   | hadoop.ozone.scm.TestContainerSmallFile |
|   | hadoop.hdds.scm.pipeline.TestSCMRestart |
|   | hadoop.ozone.client.rpc.TestBlockOutputStreamWithFailures |
|   | hadoop.ozone.om.TestOzoneManagerHA |
|   | hadoop.ozone.container.common.statemachine.commandhandler.TestBlockDeletion |
|   | hadoop.hdds.scm.pipeline.TestNode2PipelineMap |
|   | hadoop.ozone.client.rpc.TestBCSID |
|   | hadoop.ozone.web.client.TestOzoneClient |
|   | hadoop.hdds.scm.pipeline.TestPipelineClose |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.om.TestScmSafeMode |
|   | hadoop.ozone.om.TestOzoneManager |
|   | hadoop.ozone.client.rpc.TestContainerStateMachine |
|   | hadoop.ozone.TestContainerStateMachineIdempotency |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.hdds.scm.safemode.TestSCMSafeModeWithPipelineRules |
|   | hadoop.ozone.scm.TestGetCommittedBlockLengthAndPutKey |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-795/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/795 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux b23a2bee3199 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 1d70c8c |
| Default Java | 1.8.0_191 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/1/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/1/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/1/testReport/ |
| Max. process+thread count | 3503 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/client U: hadoop-hdds/client |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 50 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 399 | trunk passed |
| +1 | compile | 199 | trunk passed |
| +1 | checkstyle | 53 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 835 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 134 | trunk passed |
| 0 | spotbugs | 234 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 412 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 412 | the patch passed |
| +1 | compile | 290 | the patch passed |
| +1 | javac | 290 | the patch passed |
| +1 | checkstyle | 146 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 1162 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 325 | the patch passed |
| +1 | findbugs | 668 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 162 | hadoop-hdds in the patch failed. |
| -1 | unit | 1302 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 51 | The patch does not generate ASF License warnings. |
| | | 6795 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestCommitWatcher |
|   | hadoop.ozone.container.TestContainerReplication |
|   | hadoop.ozone.client.rpc.TestCloseContainerHandlingByClient |
|   | hadoop.ozone.client.rpc.TestReadRetries |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-795/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/795 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux db6e5c7123f2 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 597fa47 |
| Default Java | 1.8.0_191 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/2/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/2/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/2/testReport/ |
| Max. process+thread count | 3555 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/client U: hadoop-hdds/client |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/2/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 24 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 74 | Maven dependency ordering for branch |
| +1 | mvninstall | 413 | trunk passed |
| +1 | compile | 192 | trunk passed |
| +1 | checkstyle | 51 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 895 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 120 | trunk passed |
| 0 | spotbugs | 239 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 423 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 23 | Maven dependency ordering for patch |
| +1 | mvninstall | 396 | the patch passed |
| +1 | compile | 224 | the patch passed |
| +1 | javac | 224 | the patch passed |
| +1 | checkstyle | 62 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 727 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 121 | the patch passed |
| +1 | findbugs | 450 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 150 | hadoop-hdds in the patch failed. |
| -1 | unit | 1352 | hadoop-ozone in the patch failed. |
| -1 | asflicense | 34 | The patch generated 1 ASF License warnings. |
| | | 5870 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-795/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/795 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 1d638efcf0bf 4.4.0-143-generic #169~14.04.2-Ubuntu SMP Wed Feb 13 15:00:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / c4be3ea |
| Default Java | 1.8.0_191 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/3/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/3/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/3/testReport/ |
| asflicense | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/3/artifact/out/patch-asflicense-problems.txt |
| Max. process+thread count | 5417 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/client hadoop-ozone/client hadoop-ozone/ozone-manager U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/3/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

bshashikant commented:
Thanks @hanishakoneru for working on this. The patch overall looks good to me. Some minor comments:
1. BlockInputStream.java : 81 : update javadoc
2. BlockInputStream.java : 392 : correct the comment
3. It would be good to add some metrics similar to xceiverClientRatis in XceiverClientGrpc and add some tests to verify that no reads are happening as a part of seek using the metrics.

hanishakoneru commented:
Thank you @bshashikant for the review. I have addressed your review comments. 
Also, there was a mistake in bufferPosition. It should be wrt the buffers corresponding to a chunk. Fixed that too.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 51 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 2 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 63 | Maven dependency ordering for branch |
| +1 | mvninstall | 406 | trunk passed |
| +1 | compile | 201 | trunk passed |
| +1 | checkstyle | 53 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 826 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 129 | trunk passed |
| 0 | spotbugs | 240 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 422 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 27 | Maven dependency ordering for patch |
| +1 | mvninstall | 404 | the patch passed |
| +1 | compile | 209 | the patch passed |
| +1 | javac | 209 | the patch passed |
| -0 | checkstyle | 29 | hadoop-ozone: The patch generated 2 new + 0 unchanged - 0 fixed = 2 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 672 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 127 | the patch passed |
| +1 | findbugs | 433 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 160 | hadoop-hdds in the patch failed. |
| -1 | unit | 1387 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 32 | The patch does not generate ASF License warnings. |
| | | 5799 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestHybridPipelineOnDatanode |
|   | hadoop.ozone.ozShell.TestOzoneShell |
|   | hadoop.ozone.om.TestOmBlockVersioning |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestBlockOutputStream |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.om.TestMultipleContainerReadWrite |
|   | hadoop.ozone.client.rpc.TestOzoneClientRetriesOnException |
|   | hadoop.ozone.web.TestOzoneRestWithMiniCluster |
|   | hadoop.ozone.om.TestOzoneManagerHA |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.om.TestOzoneManager |
|   | hadoop.ozone.container.TestContainerReplication |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.client.rpc.TestCloseContainerHandlingByClient |
|   | hadoop.ozone.client.rpc.TestReadRetries |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-795/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/795 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 7c1109337df3 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 1b48100 |
| Default Java | 1.8.0_191 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/4/artifact/out/diff-checkstyle-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/4/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/4/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/4/testReport/ |
| Max. process+thread count | 4749 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/client hadoop-ozone/client hadoop-ozone/integration-test hadoop-ozone/ozone-manager U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/4/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

bshashikant commented:
Thanks @hanishakoneru for updating the patch. Test failures are related to the patch. Can you please check?

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 52 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 2 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 22 | Maven dependency ordering for branch |
| +1 | mvninstall | 376 | trunk passed |
| +1 | compile | 185 | trunk passed |
| +1 | checkstyle | 43 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 743 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 114 | trunk passed |
| 0 | spotbugs | 230 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 405 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 23 | Maven dependency ordering for patch |
| +1 | mvninstall | 383 | the patch passed |
| +1 | compile | 200 | the patch passed |
| +1 | javac | 200 | the patch passed |
| -0 | checkstyle | 29 | hadoop-hdds: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 651 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 126 | the patch passed |
| +1 | findbugs | 432 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 163 | hadoop-hdds in the patch failed. |
| -1 | unit | 1440 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 39 | The patch does not generate ASF License warnings. |
| | | 5602 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.ozShell.TestOzoneShell |
|   | hadoop.ozone.client.rpc.TestWatchForCommit |
|   | hadoop.hdds.scm.safemode.TestSCMSafeModeWithPipelineRules |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-795/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/795 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 3ea76763fe42 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 36267b6 |
| Default Java | 1.8.0_191 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/6/artifact/out/diff-checkstyle-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/6/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/6/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/6/testReport/ |
| Max. process+thread count | 5216 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/client hadoop-ozone/client hadoop-ozone/integration-test hadoop-ozone/ozone-manager U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/6/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 25 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 2 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 67 | Maven dependency ordering for branch |
| +1 | mvninstall | 402 | trunk passed |
| +1 | compile | 194 | trunk passed |
| +1 | checkstyle | 51 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 880 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 238 | trunk passed |
| 0 | spotbugs | 258 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 455 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 27 | Maven dependency ordering for patch |
| +1 | mvninstall | 402 | the patch passed |
| +1 | compile | 201 | the patch passed |
| +1 | javac | 201 | the patch passed |
| -0 | checkstyle | 28 | hadoop-hdds: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 735 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 118 | the patch passed |
| +1 | findbugs | 439 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 148 | hadoop-hdds in the patch failed. |
| -1 | unit | 1073 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 49 | The patch does not generate ASF License warnings. |
| | | 6641 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestBlockOutputStreamWithFailures |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-795/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/795 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 0a92c8a0bef7 4.4.0-143-generic #169~14.04.2-Ubuntu SMP Wed Feb 13 15:00:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 36267b6 |
| Default Java | 1.8.0_191 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/5/artifact/out/diff-checkstyle-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/5/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/5/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/5/testReport/ |
| Max. process+thread count | 4345 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/client hadoop-ozone/client hadoop-ozone/integration-test hadoop-ozone/ozone-manager U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/5/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 32 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 3 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 68 | Maven dependency ordering for branch |
| +1 | mvninstall | 436 | trunk passed |
| +1 | compile | 198 | trunk passed |
| +1 | checkstyle | 48 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 801 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 127 | trunk passed |
| 0 | spotbugs | 235 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 413 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 27 | Maven dependency ordering for patch |
| +1 | mvninstall | 400 | the patch passed |
| +1 | compile | 207 | the patch passed |
| +1 | javac | 207 | the patch passed |
| -0 | checkstyle | 30 | hadoop-hdds: The patch generated 9 new + 0 unchanged - 0 fixed = 9 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 663 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 126 | the patch passed |
| +1 | findbugs | 431 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 139 | hadoop-hdds in the patch failed. |
| -1 | unit | 887 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 26 | The patch does not generate ASF License warnings. |
| | | 5242 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.container.common.statemachine.commandhandler.TestCloseContainerCommandHandler |
|   | hadoop.ozone.client.rpc.TestKeyInputStream |
|   | hadoop.hdds.scm.pipeline.TestSCMPipelineManager |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-795/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/795 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 426ec512728c 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / ff27e8e |
| Default Java | 1.8.0_191 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/7/artifact/out/diff-checkstyle-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/7/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/7/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/7/testReport/ |
| Max. process+thread count | 4761 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/client hadoop-ozone/client hadoop-ozone/integration-test hadoop-ozone/ozone-manager U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/7/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

bshashikant commented:
Thanks @hanishakoneru for updating the patch. The patch looks good to me. I am +1 on this.
Please address the checkStyle issues while committing.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 856 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 4 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 60 | Maven dependency ordering for branch |
| +1 | mvninstall | 376 | trunk passed |
| +1 | compile | 189 | trunk passed |
| +1 | checkstyle | 45 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 759 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 124 | trunk passed |
| 0 | spotbugs | 238 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 414 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 26 | Maven dependency ordering for patch |
| +1 | mvninstall | 399 | the patch passed |
| +1 | compile | 211 | the patch passed |
| +1 | javac | 211 | the patch passed |
| +1 | checkstyle | 52 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 668 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 124 | the patch passed |
| +1 | findbugs | 435 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 163 | hadoop-hdds in the patch failed. |
| -1 | unit | 1344 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 37 | The patch does not generate ASF License warnings. |
| | | 6424 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.pipeline.TestRatisPipelineProvider |
|   | hadoop.ozone.client.rpc.TestOzoneClientRetriesOnException |
|   | hadoop.ozone.client.rpc.TestWatchForCommit |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-795/8/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/795 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 8529a64e9393 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 389e640 |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/8/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/8/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/8/testReport/ |
| Max. process+thread count | 5178 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/client hadoop-ozone/client hadoop-ozone/integration-test hadoop-ozone/ozone-manager U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-795/8/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hanishakoneru commented:
/retest

hanishakoneru commented:
Thank you @bshashikant . 
I fixed the checkstyle issue and run checkstyle inspection locally. Looks like CI is not working currently. I am going to merge this PR. There is no change in the last update except for checkstyle fixes. Verified locally that everything is working well.
